modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/reuse/LongListReuseBag.java

```diff
@@ -17,13 +17,13 @@
 
 package org.apache.ignite.internal.pagememory.reuse;
 
+import it.unimi.dsi.fastutil.longs.LongArrayList;
 import java.io.Externalizable;
-import org.apache.ignite.internal.util.IgniteLongList;
 
 /**
- * {@link IgniteLongList}-based reuse bag.
+ * {@link LongArrayList}-based reuse bag.
  */
-public final class LongListReuseBag extends IgniteLongList implements ReuseBag {
+public final class LongListReuseBag extends LongArrayList implements ReuseBag {
```

Comment:
Why does this class have to extend `LongArrayList`? Doesn't it make sense to use composition here?

Reply:
Also, would it make more sense to use `LongArrayFIFOQueue` here?

Reply:
Fixed it.